### ISSUE TITLE
[5.5] whereHas bindings are duplicated when using $withCount

### ISSUE THREAD

**Issue by manniL** [laravel/framework#21116]
- Laravel Version: 5.5.3
- PHP Version:  7.0.22-0ubuntu0.17.04.1
- Database Driver & Version: PDO (mysqlnd 5.0.12-dev, SQLite: 3.16.2)

### Description:

The QueryBuilder bindings are duplicated when using `whereHas()` while using `$withCount` on the related Model. This leads to problems when using MorphToMany relations, because queries like this are created through Eloquent:

![image](https://user-images.githubusercontent.com/640208/30253960-26988966-9690-11e7-8b33-67bbff15a1a4.png)

Obviously, the tipable_id is "set wrong" here because the first parameter was duplicated.

### Steps To Reproduce:

1. Setup a fresh Laravel 5.5.3 copy

2. Issue the commands `php artisan make:model Comment -m` `php artisan make:model Post -m` and `php artisan make:model Video -m`

3. Add the following code to the Comment model:

```
 public function posts()
    {
        return $this->morphedByMany(Post::class, "commentable");
    }

    public function videos()
    {
        return $this->morphedByMany(Video::class, "commentable");
    }
```

4. Add this code to the Post model:

```
 protected $withCount = ["comments"];

    public function comments()
    {
        return $this->morphToMany(Comment::class, "commentable");
    }
```
5. Now add another piece of code to the Video model:

```
public function comments()
    {
        return $this->morphToMany(Comment::class, "commentable");
    }
```

6. Finally add a migration for the commentables table through `php artisan make:migration create_commentables_table` and add those scheme definitions:

```
$table->integer('comment_id');
$table->morphs('commentable');
```

7. Create a new test by using `php artisan make:test HereIsTheBugTest`

8. Use the following snippet as the test:

```
 use RefreshDatabase;

    /**
     * A basic test example.
     *
     * @return void
     */
    public function testThatItFails()
    {
        $p = Post::create();
        $v = Video::create();
        $c = new Comment();

        $c->save();
        $c->posts()->save($p);
        $c->videos()->save($v);

        $this->assertTrue(Comment::first()->is($c));

        $testPost = Post::firstOrFail();
        $this->assertTrue($testPost->is($p));

        $builder = Comment::whereHas("posts");

        $this->assertTrue($c->is($builder->firstOrFail()));
    }
```

9. Setup your phpunit.xml to use a testing in-memory database by adding:

```
        <env name="DB_CONNECTION" value="sqlite"/>
        <env name="DB_DATABASE" value=":memory:"/>
```

10. Don't forget to fix your database.php file (see [here](https://github.com/laravel/laravel/commit/80e96d276a5c232705ed62dfd55dec50e6030c8c))

11. Run the test and see it fail!

```
1) Tests\Feature\HereIsTheBugTest::testThatItFails
Illuminate\Database\QueryException: SQLSTATE[HY000]: General error: 25 bind or column index out of range (SQL: select * from "comments" where exists (select * from "posts" inner join "commentables" on "posts"."id" = "commentables"."commentable_id" where "comments"."id" = "commentables"."comment_id" and "commentables"."commentable_type" = App\Post) limit 1)
```

### Related issues

#20640

**Comment by Dylan-DPC** [comment]
The problem isn't with `withCount` but with `whereHas`. This is a known problem (ref #5429 #13585 #18523)

**Comment by manniL** [comment]
@Dylan-DPC `whereHas` is only a problem in combination with polymorphic relations when the relations aren't strictly defined, which isn't the case here. As you can see in the example app, `whereHas` works well when you have two relation functions (`posts` for all entries that have the type `App\Models\Post` and `videos` for all video entries).

If you remove the `$withCount`variable of the `Post` Model inside the test application you can see that the test I've created passes.

**Comment by themsaid** [comment]
Please include code in your steps of replication, including an entire laravel application in a .zip file won't help :)

What code should I run in a fresh laravel project to be able to replicate?

**Comment by manniL** [comment]
@themsaid  I've updated the information concerning that issue and added the whole code for reproduction!

**Comment by Dylan-DPC** [comment]
It is better to push the repo on GH and share the link. That way people can clone it and try it directly. Works better than the zip :+1:

**Comment by manniL** [comment]
You are right @Dylan-DPC 🙈 Why didn't I think of that before? 😅

**Comment by brunogaspar** [comment]
Unfortunately i have this issue as well.

Seems like this `$withCount` property feature was not implemented properly :\

**Comment by manniL** [comment]
Looks like this issue is a duplicate of #21464 
Fortunately, there is a fix incoming #21468
